test/src/test/java/jenkins/cli/StopBuildsCommandTest.java

```diff
@@ -0,0 +1,217 @@
+                        "No builds stopped" + LN));
+    }
+
+    @Test
+    public void shouldStopSecondJobEvenIfFirstStopFailed() throws Exception {
+        final FreeStyleProject project = createLongRunningProject(TEST_JOB_NAME_2);
+
+        final FreeStyleProject restrictedProject = createLongRunningProject(TEST_JOB_NAME);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(new MockAuthorizationStrategy().
+                grant(Jenkins.READ).everywhere().toEveryone().
+                grant(Item.READ).onItems(restrictedProject, project).toEveryone().
+                grant(Item.CANCEL).onItems(restrictedProject).toAuthenticated().
+                grant(Item.CANCEL).onItems(project).toEveryone());
+
+        restrictedProject.scheduleBuild2(0).waitForStart();
+        project.scheduleBuild2(0).waitForStart();
+
+        final String stdout = runWith(asList(TEST_JOB_NAME, TEST_JOB_NAME_2)).stdout();
+
+        assertThat(stdout,
+                equalTo("Exception occurred while trying to stop build '#1' for job 'jobName'. " +
+                        "Exception class: AccessDeniedException2, message: anonymous is missing the Job/Cancel permission" + LN +
+                        "Build '#1' stopped for job 'jobName2'" + LN));
+    }
+
+    private CLICommandInvoker.Result runWith(final List<String> jobNames) throws Exception {
+        CLICommand cmd = new StopBuildsCommand();
+        CLICommandInvoker invoker = new CLICommandInvoker(j, cmd);
+        return invoker.invokeWithArgs(jobNames.toArray(new String[jobNames.size()]));
+    }
+
+    private void setupAndAssertTwoBuildsStop(final List<String> inputNames) throws Exception {
+        setupSlaveWithTwoExecutors();
+
+        final FreeStyleProject project = createLongRunningProject(TEST_JOB_NAME);
+        final FreeStyleProject project2 = createLongRunningProject(TEST_JOB_NAME_2);
+
+        project.scheduleBuild2(0).waitForStart();
+        project2.scheduleBuild2(0).waitForStart();
+
+        final String stdout = runWith(inputNames).stdout();
+
+        assertThat(stdout, equalTo("Build '#1' stopped for job 'jobName'" + LN +
+                "Build '#1' stopped for job 'jobName2'" + LN));
+
+        waitForLastBuildToStop(project);
+        waitForLastBuildToStop(project2);
+    }
+
+    private FreeStyleProject createLongRunningProject(final String jobName) throws IOException {
+        final FreeStyleProject project = j.createFreeStyleProject(jobName);
+        project.getBuildersList().add(createScriptBuilder("sleep 50000"));
+        return project;
+    }
+
+    private void setupSlaveWithTwoExecutors() throws hudson.model.Descriptor.FormException, IOException, URISyntaxException {
+        DumbSlave slave = new DumbSlave("slave", j.jenkins.getRootDir().getAbsolutePath(), j.createComputerLauncher(null));
+        slave.setNumExecutors(2);
```

Comment:
Oddly enough, this is the default value already.

Reply:
You right!
`setupSlaveWithTwoExecutors()` calls unnecessary 
Will fix that

Reply:
Fixed in new PR https://github.com/jenkinsci/jenkins/pull/4013